### PR TITLE
WRQ-27790: Updated node modules major versions

### PR DESCRIPTION
**Pull request by paul-beldean-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Updated node modules major versions

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Updated node modules major versions

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
Replaced "getCLS, getFID, getFCP, getLCP, getTTFB" with "onCLS, onFID, onFCP, onLCP, onTTFB" because they were deprecated and replaced with the new ones.
Replaced "ReportHandler" type with "(metric: MetricType) => void" because it was deprecated.
Removed SlotItem and ToggleItem as ui/SlotItem and ui/ToggleItem were removed from enact/ui 5.0.0 and removed CheckboxItem and RadioItem because they were using ToggleItem.
Added node engine version to electron template.

Change logs:
web-vitals v4.0.0
https://github.com/GoogleChrome/web-vitals/blob/main/CHANGELOG.md#v400-2024-05-13
Update types to support more generic usage
Split waitingDuration to make it easier to understand redirect delays
Rename TTFBAttribution fields from *Time to *Duration
Rename resourceLoadTime to resourceLoadDuration in LCP attribution
Add INP breakdown timings and LoAF attribution
Deprecate onFID() and remove previously deprecated APIs
Expose the target element in INP attribution
Save INP target after interactions to reduce null values when removed from the DOM
Cap TTFB in attribution
Fix reportAllChanges behavior for LCP when library is loaded late
upgrading: https://github.com/GoogleChrome/web-vitals/blob/main/docs/upgrading-to-v4.md

electron v31.0.0
https://github.com/electron/electron/releases/tag/v31.0.0
Made window.flashFrame(bool) flash continuously on macOS.
Updated Chromium to 126.0.6445.0. (Removed: WebSQL support)
Updated Chromium to 125.0.6412.0. (Behavior Changed: nativeImage.toDataURL will preserve PNG colorspace)

ramda v0.30.0
https://github.com/ramda/ramda/issues/3453
use === to test for placeholder

### Links
[//]: # (Related issues, references)
WRQ-27790

### Comments

Enact-DCO-1.0-Signed-off-by: Paul Beldean paul.beldean@lgepartner.com

### CODE REVIEW

File: packages/agate/template/src/reportWebVitals.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
 const reportWebVitals = (onPerfEntry) => {
 	if (onPerfEntry && onPerfEntry instanceof Function) {
-		import('web-vitals').then(({getCLS, getFID, getFCP, getLCP, getTTFB}) => {
-			getCLS(onPerfEntry);
-			getFID(onPerfEntry);
-			getFCP(onPerfEntry);
-			getLCP(onPerfEntry);
-			getTTFB(onPerfEntry);
+		import('web-vitals').then(({onCLS, onFID, onFCP, onLCP, onTTFB}) => {
```
Comment (daniel-stoian-lgp) [inline packages/agate/template/src/reportWebVitals.js]: can you please also add the function onINP  for the new INP metric on all templates?

**Comment by daniel-stoian-lgp** [comment]
the web-vitals upgrade for moonstone template is missing. please add it

**Review by daniel-stoian-lgp — APPROVED** [review]

**Comment by hong6316** [comment]
- Please revert adding `UiSlotItem` and `UiToggleItem` and remove `SlotItem` and `ToggleItem`.
Enact v5 removed them so `packages/theme` also should follow those changes.
- In `Additional Considerations`, Please add major-updated modules'(`web-vitals`, `electron`, and `ramda`) change logs, includes the migration guide.

**Comment by paul-beldean-lgp** [comment]
> * Please revert adding `UiSlotItem` and `UiToggleItem` and remove `SlotItem` and `ToggleItem`.
>   Enact v5 removed them so `packages/theme` also should follow those changes.
> * In `Additional Considerations`, Please add major-updated modules'(`web-vitals`, `electron`, and `ramda`) change logs, includes the migration guide.

I made the changes. For electron v31 there is no upgrade guide.

**Review by hong6316 — DISMISSED** [review]
👍 

But please add minimun node requirement in electron template.
When I checked the electron changelog, node version was updated.
https://github.com/electron/electron/releases/tag/v31.0.0
```javascript
  "engines": {
    "node": ">= 20.14.0"
  },
```

**Review by hong6316 — APPROVED** [review]
👍

**Pull request merged**